apps/api/v1/test/lib/bookings/_auth-middleware.integration-test.ts

```diff
@@ -0,0 +1,93 @@
+import type { Request, Response } from "express";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { createMocks } from "node-mocks-http";
+import { describe, it, expect } from "vitest";
+
+import type { HttpError } from "@calcom/lib/http-error";
+import prisma from "@calcom/prisma";
+
+import authMiddleware from "../../../pages/api/bookings/[id]/_auth-middleware";
+
+type CustomNextApiRequest = NextApiRequest & Request;
+type CustomNextApiResponse = NextApiResponse & Response;
+
+describe("Bookings auth middleware", () => {
+  it("Returns 403 when user has no permission to the booking", async () => {
+    const trialUser = await prisma.user.findFirstOrThrow({ where: { email: "trial@example.com" } });
```

Comment:
Note: ideally we don’t directly use prisma here. Will check to see if a reusable repository is available. 

Note: instead of hardcoding the emails of users, would like to build out utils that defined these built-in test users.